toogoodtogo_ha_mqtt_bridge/main.py

```diff
@@ -228,10 +276,31 @@ def on_disconnect(client, userdata, rc):
         client.reconnect()
 
 
+def calc_timeout():
+    global watchdog_timeout
+    now = datetime.now()
+    tgtg = settings.get("tgtg")
+
+    if "polling_schedule" not in tgtg:
+        logger.error("No polling_schedule found in settings")
+        os._exit(1)
+
+    cron_schedule = tgtg.polling_schedule
+    if croniter.is_valid(cron_schedule):
+        next_run = croniter(cron_schedule, now).get_next(datetime)
+        for i in range(2):
+            next_run = croniter(cron_schedule, next_run).get_next(datetime)
+        watchdog_timeout = (next_run - now).seconds + 10  # Offset
```

Comment:
Shouldn't we use the `tgtg_client.timeout` at that point? 
Could brevent nasty corner cases like 2 failed pulls and one that takes a little longer than 10 seconds?

Reply:
Good hint, I will change this.

Reply:
Done with commit [140cbce](https://github.com/MaxWinterstein/toogoodtogo-ha-mqtt-bridge/pull/28/commits/140cbce38d1e36bc4e65b7f5cf0ce62ce8009292)